core/src/main/java/io/opentelemetry/android/OpenTelemetryRum.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android
+
+import android.app.Application
+import io.opentelemetry.android.config.OtelRumConfig
+import io.opentelemetry.android.session.SessionProvider
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.sdk.OpenTelemetrySdk
+
+/**
+ * Entrypoint for the OpenTelemetry Real User Monitoring library for Android.
+ */
+interface OpenTelemetryRum {
+    /**
+     * Get a handle to the instance of the OpenTelemetry API that this
+     * instance is using for instrumentation.
+     */
+    val openTelemetry: OpenTelemetry
+
+    /**
+     * Get the client session ID associated with this instance of the RUM instrumentation library.
+     * Note: this value will change throughout the lifetime of an application instance, so it is
+     * recommended that you do not cache this value, but always retrieve it from here when needed.
+     */
+    fun getRumSessionId(): String
```

Comment:
Isn't it more consistent to also implement this as a `val` like the `openTelemetry` above?

Reply:
+1 on the consistency. I actually prefer having all functions on interfaces, even though there's no reason why not using `val` for Kotlin interfaces, maybe I'm just old 😅, though I do like the idea of keeping consistency in general.